File: solutions/go/binary-search/2/binary_search.go
package binarysearch

func SearchInts(list []int, key int) int {
	left := 0
	right := len(list) - 1

	for left <= right {
		midpoint := (left + right) / 2

		if list[midpoint] > key {
			right = midpoint - 1
		} else if list[midpoint] < key {
			left = midpoint + 1
		} else {
			return midpoint
		}
	}

	return -1
}

File: solutions/zig/binary-search/1/binary_search.zig
// Take a look at the tests, you might have to change the function arguments

pub fn binarySearch(comptime T: type, target: T, items: []const T) ?usize {
    var left: usize = 0;
    var right: usize = items.len;

    while (left < right) {
        const mid: usize = (left + right) / 2;

        if (items[mid] > target) {
            right = mid;
        } else if (items[mid] < target) {
            left = mid + 1;
        } else {
            return mid;
        }
    }

    return null;
}
